vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -681,13 +683,49 @@ public void remove(Component... components) {
         /**
          * Removes all components from the container.
          */
+        @Override
         public void removeAll() {
             root.removeAllChildren();
             dialog.getElement()
                     .executeJs("this." + rendererFunction + " = null;");
             setRendererCreated(false);
         }
 
+        /**
+         * Adds the given component to the container at the specific index.
+         *
+         * @param index
+         *            the index, where the component will be added. The index
+         *            must be non-negative and may not exceed the children count
+         *            component – the component to add, value should not be null
+         * @param component
+         *            the component to be added.
+         */
+        @Override
+        public void addComponentAtIndex(int index, Component component) {
+            Objects.requireNonNull(component, "Component should not be null");
+            if (index < 0) {
+                throw new IllegalArgumentException(
+                        "Cannot add a component with a negative index");
+            } else {
+                root.insertChild(index, component.getElement());
+            }
```

Comment:
Can we not call `HasComponents.addComponentAtIndex` here to reuse it? Same with all the other methods.

Reply:
Reused the default methods in all overridden methods.